mono/mini/interp/interp.c

```diff
@@ -413,6 +413,8 @@ interp_free_context (gpointer ctx)
 {
 	ThreadContext *context = (ThreadContext*)ctx;
 
+	set_context (NULL);
```

Comment:
I don't like that we move from freeing whatever context to assuming that this is the context for the current thread. Since this method is now making some implicit assumptions about the caller logic, I think we should at least assert that this context is the context for the current thread, before clearing it from TLS.